modules/sort/main/index.coffee

```diff
@@ -1,36 +1,56 @@
 # Intl.Collator isn't supported by safari
 # https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Collator
 # using a collator is supposed to be faster than doing localeCompare
-hasCollator = Intl?.Collator?
+hasCollator = -> Intl?.Collator?
 
-collator = if hasCollator
+hx._.sort = {}
+
+collatorFn = -> if hasCollator()
   new Intl.Collator(undefined, {numeric: true}).compare
 else
   (a, b) ->
-    if a < b then -1
-    else if a > b then 1
-    else 0
+    if a is b then 0
+    else if String(a) < String(b) then -1
+    else 1
 
-compare = (a, b) ->
-  if not isNaN(Number(a)) and not isNaN(Number(b)) then a - b
+nullsLastCollator = (collator) -> (a, b) ->
+  if a is b then 0
+  else if a is undefined then 1
+  else if b is undefined then -1
+  else if a is null then 1
+  else if b is null then -1
+  else if not isNaN(Number(a)) and not isNaN(Number(b)) then a - b
```

Comment:
`else defaultCollator(collator)`? It probably doesn't matter that the default collator also checks `a? and b?`